plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1677,6 +1679,85 @@ public void testTableWithTrailingSlashLocation(boolean partitioned)
+        assertUpdate("INSERT INTO " + tableName + " VALUES('url3', 'domain3', 3)", 1);
+        assertQueryReturnsEmptyResult("SELECT * FROM TABLE(system.table_changes(" +
+                "schema_name =>'test_schema', table_name => '" + tableName + "', since_timestamp => TIMESTAMP '2024-10-31 01:00'))");
+
+        assertUpdate("UPDATE " + tableName + " SET page_url = 'url22' WHERE domain = 'domain2'", 1);
+        assertUpdate("UPDATE " + tableName + " SET page_url = 'url33' WHERE views = 3", 1);
+        assertUpdate("DELETE FROM " + tableName + " WHERE page_url = 'url1'", 1);
+
+        assertQuery("SELECT * FROM " + tableName,
+                """
+                 VALUES
+                    ('url22', 'domain2', 2),
+                    ('url33', 'domain3', 3)
+                 """);
+
+        assertTableChangesQuery("SELECT * FROM TABLE(system.table_changes(" +
+                "schema_name =>'test_schema', table_name => '" + tableName + "', since_timestamp => TIMESTAMP '2022-10-31 01:00'))",
+                """
+                        VALUES
+                            ('url1', 'domain1', 1, 'insert', BIGINT '1'),
+                            ('url2', 'domain2', 2, 'insert', BIGINT '2'),
+                            ('url3', 'domain3', 3, 'insert', BIGINT '3'),
+                            ('url2', 'domain2', 2, 'update_preimage', BIGINT '4'),
+                            ('url22', 'domain2', 2, 'update_postimage', BIGINT '4'),
+                            ('url3', 'domain3', 3, 'update_preimage', BIGINT '5'),
+                            ('url33', 'domain3', 3, 'update_postimage', BIGINT '5'),
+                            ('url1', 'domain1', 1, 'delete', BIGINT '6')
+                        """);
+
+        assertTableChangesQuery("SELECT * FROM TABLE(system.table_changes('test_schema', '" + tableName + "'))",
+                """
+                        VALUES
+                            ('url1', 'domain1', 1, 'insert', BIGINT '1'),
+                            ('url2', 'domain2', 2, 'insert', BIGINT '2'),
+                            ('url3', 'domain3', 3, 'insert', BIGINT '3'),
+                            ('url2', 'domain2', 2, 'update_preimage', BIGINT '4'),
+                            ('url22', 'domain2', 2, 'update_postimage', BIGINT '4'),
+                            ('url3', 'domain3', 3, 'update_preimage', BIGINT '5'),
+                            ('url33', 'domain3', 3, 'update_postimage', BIGINT '5'),
+                            ('url1', 'domain1', 1, 'delete', BIGINT '6')
+                        """);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS");
+        List<String> timestamps = getQueryRunner().execute("SELECT timestamp FROM \"" + tableName + "$history\" ORDER BY version ASC")
+                .getOnlyColumn()
+                .map(column -> ((ZonedDateTime) column).withZoneSameInstant(ZoneId.of("UTC")).toLocalDateTime().format(formatter))
```

Comment:
add "TIMESTAMP '...'" here so that queries below are simpler (you can do this by configuring DateTimeFormatter appropriately, see `io.trino.plugin.deltalake.TestDeltaLakeReadTimestamps#WHERE_VALUE_FORMATTER`

add zone "UTC" (explained below)